onnxruntime/contrib_ops/cuda/collective/sharded_moe.h

```diff
@@ -27,7 +27,9 @@ class ShardedMoE final : public NcclKernel, public MoEBase {
 
   int64_t local_experts_start_index_;
   int64_t tensor_shards_;
-  std::vector<int64_t> rank_to_experts_start_index_;
+  mutable std::vector<int64_t> rank_to_experts_start_index_;
+
+  mutable ort_trtllm::GlobalIPCMemoryResourcePack g_ipc_mem_res_pack_;
 };
```

Comment:
This creates a thread safety issue. Is this protected by a mutex when inference is invoked from multiple threads?

Reply:
it's missed. I just atomized the GetWorkspace() + deserialize()

Reply:
Please, add a comment that explains it.